### PR TITLE
Add shared functionality to get the parsed kernel cmdline

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-kernel.c [hunk 1]
```diff
@@ -218,3 +219,40 @@ fu_kernel_reset_firmware_search_path(GError **error)
...
+ * @error: (nullable): optional return location for an error
+ *
+ * Loads all the kernel /proc/cmdline key/values into a hash table.
+ *
+ * Returns: (transfer container) (element-type utf8 utf8): keys from the kernel command line
+ *
+ * Since: 1.8.5
+ **/
+GHashTable *
+fu_kernel_get_proc_cmdline(GError **error)
```
Comment (superm1) [inline libfwupdplugin/fu-kernel.c:234]: I think this should have some `#ifndef _WIN32` guard

Comment (hughsie) [reply libfwupdplugin/fu-kernel.c:234]: Does macos have /proc/cmdline? The one in fu-engine.c has #ifdef __linux__

Comment (superm1) [reply libfwupdplugin/fu-kernel.c:234]: I'm pretty sure `/proc/cmdline` is a linux only concept.
Windows has https://docs.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getcommandlinea, but that's not really the same thing as `/proc/cmdline`, it's more like `/proc/self/cmdline`.

Comment (hughsie) [reply libfwupdplugin/fu-kernel.c]: Gotcha, thanks. Fixed.

**Review by superm1 — APPROVED** [review]

**Comment by DylanVanAssche** [comment]
Will use this in #4993 once merged.

**Comment by hughsie** [comment]
Done!

**Review by DylanVanAssche — APPROVED** [review]
LGTM

**Pull request merged**